core/src/main/scala/akka/kafka/Metadata.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka
+
+import java.util.Optional
+
+import akka.actor.NoSerializationVerificationNeeded
+import org.apache.kafka.clients.consumer.{OffsetAndMetadata, OffsetAndTimestamp}
+import org.apache.kafka.common.{PartitionInfo, TopicPartition}
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+/**
+ * Messages for Kafka metadata fetching via [[KafkaConsumerActor]].
+ *
+ * NOTE: Processing of these requests blocks the actor loop. The KafkaConsumerActor is configured to run on its
+ * own dispatcher, so just as the other remote calls to Kafka, the blocking happens within a designated thread pool.
+ * However, calling these during consuming might affect performance and even cause timeouts in extreme cases.
+ */
+object Metadata {
+
+  sealed trait Request
+  sealed trait Response
+
+  /**
+   * [[org.apache.kafka.clients.consumer.KafkaConsumer#listTopics()]]
+   */
+  case object ListTopics extends Request with NoSerializationVerificationNeeded
+  final case class Topics(response: Try[Map[String, List[PartitionInfo]]]) extends Response with NoSerializationVerificationNeeded {
+    /**
+     * Java API
+     */
+    def getResponse: Optional[java.util.Map[String, java.util.List[PartitionInfo]]] =
+      response.map { m => Optional.of(m.mapValues(_.asJava).asJava) }.getOrElse(Optional.empty())
+  }
+
+  /**
+   * Java API:
+   * [[org.apache.kafka.clients.consumer.KafkaConsumer#listTopics()]]
+   */
+  val listTopics: ListTopics.type = ListTopics
+
+  /**
+   * [[org.apache.kafka.clients.consumer.KafkaConsumer#partitionsFor()]]
+   */
+  final case class GetPartitionsFor(topic: String) extends Request with NoSerializationVerificationNeeded
```

Comment:
Add `create(...)` methods for the Java API for this request message and others below.